File: livingdoc-fixture-api/src/main/java/org/livingdoc/fixture/api/converter/common/StringConverter.java
package org.livingdoc.fixture.api.converter.common;

import java.lang.reflect.AnnotatedElement;

import org.livingdoc.fixture.api.converter.TypeConverter;


public class StringConverter implements TypeConverter<String> {

    @Override
    public String convert(String value, AnnotatedElement element) {
        return value;
    }

}

File: livingdoc-fixture-converter/src/main/kotlin/org/livingdoc/fixture/converter/StringConverter.kt
package org.livingdoc.fixture.converter

import org.livingdoc.fixture.api.converter.TypeConverter
import java.lang.reflect.AnnotatedElement


open class StringConverter : TypeConverter<String> {

    override fun convert(value: String, element: AnnotatedElement?) = value

    override fun canConvertTo(targetType: Class<*>): Boolean {
        val isJavaObjectType = String::class.javaObjectType == targetType
        val isKotlinType = String::class.java == targetType
        return isJavaObjectType || isKotlinType
    }

}
